src/main/java/jcifs/smb/NtlmPasswordAuthenticator.java

```diff
@@ -0,0 +1,640 @@
+                char c = userInfo.charAt(i);
+                if ( c == ';' ) {
+                    dom = userInfo.substring(0, i);
+                    u = i + 1;
+                }
+                else if ( c == ':' ) {
+                    pass = userInfo.substring(i + 1);
+                    break;
+                }
+            }
+            user = userInfo.substring(u, i);
+        }
+
+        this.domain = dom != null ? dom : ( defDomain != null ? defDomain : "" );
+        this.username = user != null ? user : ( defUser != null ? defUser : "" );
+        this.password = pass != null ? pass : ( defPassword != null ? defPassword : "" );
+
+        if ( type == null ) {
+            this.type = guessAuthenticationType();
+        }
+        else {
+            this.type = type;
+        }
+    }
+
+
+    /**
+     * @return
+     */
+    protected AuthenticationType guessAuthenticationType () {
+        AuthenticationType t = AuthenticationType.USER;
+        if ( "guest".equalsIgnoreCase(this.username) ) {
+            t = AuthenticationType.GUEST;
+        }
+        else if ( ( getUserDomain() == null || getUserDomain().isEmpty() ) && getUsername().isEmpty() && ( getPassword().isEmpty() ) ) {
+            t = AuthenticationType.NULL;
+        }
+        return t;
+    }
+
+
+    @SuppressWarnings ( "unchecked" )
+    @Override
+    public <T extends Credentials> T unwrap ( Class<T> t ) {
+        if ( t.isAssignableFrom(this.getClass()) ) {
+            return (T) this;
+        }
+        return null;
+    }
+
+
+    @Override
+    public Subject getSubject () {
+        return null;
+    }
+
+
+    @Override
+    public void refresh () throws CIFSException {}
+
+
+    /**
+     * 
+     * {@inheritDoc}
+     *
+     * @see jcifs.smb.CredentialsInternal#createContext(jcifs.CIFSContext, java.lang.String, java.lang.String, byte[],
+     *      boolean)
+     */
+    @Override
+    public SSPContext createContext ( CIFSContext tc, String targetDomain, String host, byte[] initialToken, boolean doSigning ) throws SmbException {
+        if ( tc.getConfig().isUseRawNTLM() ) {
+            return setupTargetName(tc, host, new NtlmContext(tc, this, doSigning));
+        }
+
+        try {
+            if ( initialToken != null && initialToken.length > 0 ) {
+                NegTokenInit tok = new NegTokenInit(initialToken);
+                if ( log.isDebugEnabled() ) {
+                    log.debug("Have initial token " + tok);
+                }
+                if ( tok.getMechanisms() != null ) {
+                    Set<ASN1ObjectIdentifier> mechs = new HashSet<>(Arrays.asList(tok.getMechanisms()));
+                    if ( !mechs.contains(NtlmContext.NTLMSSP_OID) ) {
+                        throw new SmbUnsupportedOperationException("Server does not support NTLM authentication");
+                    }
+                }
+            }
+        }
+        catch ( SmbException e ) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: An exception is caught and re-thrown within the catch clause without any other operations being performed on the exception, making the catch clause unnecessary.

**Fix**: Perform additional operations on the exception that has been caught. You can wrap a standard exception into a custom exception to add additional information or special handling. When you do that, make sure to set the original exception as the cause. Otherwise, you will lose the stack trace and message of the original exception, which will make it difficult to analyze the event that caused the exception.  
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)